src/test/crawl-url.js

```
@@ -0,0 +1,380 @@
+    
+    if (isLikelyCompressed || buffer.length > 100) {
+      console.log('Content-encoding not specified, but content might be compressed. Trying common formats...');
+      
+      // Try gzip
+      try {
+        console.log('Attempting gzip decompression...');
+        const gunzip = util.promisify(zlib.gunzip);
+        const gzipResult = await gunzip(buffer);
+        console.log(`Gzip decompression successful: ${gzipResult.length} bytes`);
+        decompressedContent = gzipResult;
+      } catch (gzipError) {
```

ESLint

[error] 'gzipError' is defined but never used.

(no-unused-vars)